03/app.js

```diff
@@ -3,29 +3,64 @@ import ReactDOM from 'react-dom';
 
 class Article extends React.Component {
     state = {
-        comments: [],
+        comments: [
+            {
+                text: 'some comment 1',
+                id: 1
+            }, 
+            {
+                text: 'some comment 2',
+                id: 2
+            }, 
+        ],
+        commentInput: ''
     }
     
+    handleSubmit = event => {
+        event.preventDefault();
+        this.setState({
+            comments: this._addComment(),
+            commentInput: ''
+        })
+    }
+    _addComment = () => {
+        return this.state.comments.concat({
+            text: this.state.commentInput,
+            id: this._getId()
+        })
+    }
+    _getId = () => this.state.comments.reduce((acc, comment) => acc = acc > comment.id ? acc : comment.id) + 1;
```

Comment:
raczej bez `acc = ` (1) oraz dodanie wartości początkowej (2) czyli `_getId = () => this.state.comments.reduce((acc, comment) => acc > comment.id ? acc : comment.id, 0) + 1;`

ad 1) nie powinniśmy modyfikować parametrów
ad 2) nie będzie działać dla pustej tablicy

Reply:
OK poprawione!